types/underscore/underscore-tests.ts

```diff
@@ -2978,6 +2978,14 @@ _.defer(() => alert("deferred")); // $ExpectType void
     welcome("moe"); // $ExpectType any
 }
 
+// restArguments
+
+// Used to retrieve all parameters of a function when called, except for the first parameter, and return them as an array.
+{
+    const concat = _.restArguments((...str: string[]) => str.join(' '), startIndex);
+    concat(('Hello', 'World', '!'), 0);
```

Comment:
This is not how `_.restArguments` works.

Suppose that you have a function like this:

```ts
const unwrapped = (words: string[]) => words.join(' ');
```

then `_.restArguments(unwrapped)` will return a function that behaves like this variant:

```ts
const wrapped = (...words: string[]) => words.join(' ');
```

The `startIndex` is an argument that you pass to `_.restArguments`, **not** to the `wrapped` function that it returns like you did with the `0` above. You also passed a `startIndex` name to `_.restArguments` above, but that is an undefined variable.